File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        # initilise the num set
        numSet = set(nums)
        longest_seq  = 0
        for n in numSet:

            if (n - 1) not in numSet:
                length = 0
                while (n + length) in numSet:
                    length += 1
                longest_seq = max(length, longest_seq)

        return longest_seq            

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.dart
class Solution {
   int longestConsecutive(List<int> nums) {
        
        Set<int> numSet = nums.toSet();

        int longest = 0;

        for(int n in nums) {

            if(!numSet.contains(n-1)) {

                int current_length = 0;

                while(numSet.contains(n + current_length)) {
                    current_length ++;
                }

                longest = max(current_length, longest);
            }
        }
    return longest;    
  }
}